docs/developer-docs/integrations/exchange-rate/exchange-rate-canister.md

```diff
@@ -0,0 +1,134 @@
+# Exchange rate canister
+
+## Overview
+
+The exchange rate canister, referred to as the XRC, is a canister that runs on the **uzr34 system subnet**. The XRC uses [HTTPS outcalls](https://internetcomputer.org/https-outcalls/) to fetch data from major cryptocurrency exchanges by using the exchange's public API to retrieve real time or historical cryptocurrency pricing information. The XRC also queries the public APIs for foreign exchange data providers around the world periodically in order to get forex rates. 
+
+The XRC canister can be integrated with decentralized exchanges (DEXs) to provide functionality such as comparing exchange rates against market rates to determine the value of assets that are held in a canister smart contract. For example, the cycle minting canister of the NNS uses the XRC to obtain the current ICP/XDR exchange rates, which it requires for the conversion of ICP to cycles.
+
+The XRC provides this data in response to requests made by other canisters. A request made by another canister is composed of a base asset, a quote asset, and an optional UNIX epoch timestamp. 
+
+An **asset** is a record consisting of a symbol (such as "ICP") and a class (either 'Cryptocurrency' or 'FiatCurrency'). The base and quote asset can be any combination of cryptocurrency and fiat currency assets, such as ICP/USD, USD/EUR, BTC/ICP, etc. The timestamp parameter enables the functionality for historic rates to be requested. If no timestamp is provided, the exchange rate for the current time is returned by default. 
+
+## Usage
+The canister ID of the XRC is `uf6dk-hyaaa-aaaaq-qaaaq-cai`. 
+
+The exchange rate canister offers a single endpoint:
+
+```
+"get_exchange_rate": (GetExchangeRateRequest) -> (GetExchangeRateResult)
+```
+
+To make a request to the XRC, the following request form can be used: 
+```
+type GetExchangeRateRequest = record {
+   base_asset: Asset;
+   quote_asset: Asset;
+   timestamp: opt nat64;
+}; 
+```
+
+Once this form has been sent to the XRC, the following response is returned:
+
+```
+type GetExchangeRateResult = variant {
+   Ok: ExchangeRate;
+   Err: ExchangeRateError;
+};
+```
+
+## Cycles cost 
+Each request to the XRC costs 10B cycles to submit the request, otherwise an `ExchangeRateError::NotEnoughCycles` error will be returned. The actual cycles cost of the request call will depend on two factors: the requested asset types and the state of the internal exchange rate cache. These parameters are as follows:
```

Comment:
Note that the prices will change soon! They will be reduced by a factor of 10 in the next upgrade, which should happen around July 24.